components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/spark/algorithms/SupervisedSparkModelBuilder.java

```diff
@@ -167,6 +176,15 @@ public MLModel build() throws MLModelBuilderException {
         }
     }
 
+    private String getTypeOfResponseVariable(String responseVariable, List<Feature> features){
+        String type = null;
+        for(Feature feature: features){
+            if(feature.getName().equals(responseVariable))
+                type = feature.getType();
+        }
+        return type;
```

Comment:
This method could return null right? Then please handle it properly in the place it is used.